packages/firestore/src/api/database.ts

```diff
@@ -413,6 +419,32 @@ export class Firestore implements PublicFirestore, FirebaseService {
     }
   }
 
+  useEmulator(host: string, port: number): void {
+    validateExactNumberOfArgs('Firestore.useEmulator', arguments, 2);
+    validateArgType('Firestore.useEmulator', 'string', 1, host);
+    validateArgType('Firestore.useEmulator', 'number', 2, port);
+
+    if (this._firestoreClient) {
+      throw new FirestoreError(
+        Code.FAILED_PRECONDITION,
+        'Firestore has already been started and its settings can no longer be changed. ' +
+          'You can only call useEmulator() before calling any other methods on a Firestore object.'
+      );
+    }
```

Comment:
Do we need to have this check both here and in `settings`? Seems like we could get away with one check if we make the error message a little bit more vague (or just dropping the second sentence).

Reply:
No I was just trying to have distinct error messages, I'll merge it.